great_expectations/cli/cli.py

```diff
@@ -156,6 +171,14 @@ def cli(ctx, v3_api, verbose, config_file_location, assume_yes):
     else:
         cli_message("Using v2 (Batch Kwargs) API")
 
+        ge_config_version: float = (
+            ctx.obj.get_data_context_from_config_file().get_config().config_version
+        )
+        if ge_config_version >= FIRST_GE_CONFIG_VERSION_WITH_CHECKPOINT_STORE:
+            raise ge_exceptions.InvalidDataContextConfigError(
+                f"Using the legacy v2 (Batch Kwargs) API with a recent config version ({ge_config_version}) is illegal."
+            )
```

Comment:
❤️

Reply:
NICE! Much better error messsage. Thank you.